src/main/java/edu/ie3/datamodel/io/source/sql/SqlWeatherSource.java

```diff
@@ -62,26 +63,38 @@ public SqlWeatherSource(
       TimeBasedWeatherValueFactory weatherFactory) {
     this.connector = connector;
     this.idCoordinateSource = idCoordinateSource;
-    this.coordinateColumnName = weatherFactory.getCoordinateIdFieldString();
     this.weatherFactory = weatherFactory;
+    this.factoryCoordinateFieldName = weatherFactory.getCoordinateIdFieldString();
 
-    String timeColumnName = weatherFactory.getTimeFieldString();
+    String dbTimeColumnName =
+        dbColumnName(weatherFactory.getTimeFieldString(), connector, weatherTableName)
```

Comment:
Possibly introduce attribute for `weatherFactory.getTimeFieldString()` as well?

Reply:
IMHO not required here, as the second call only takes place on error. Hence in normal operation there is only one call required which does not need an attribute here.